src/pybind/mgr/object_format.py

```diff
@@ -1,5 +1,133 @@
 # object_format.py provides types and functions for working with
 # requested output formats such as JSON, YAML, etc.
+"""tools for writing formatting-friendly mgr module functions
+
+Currently, the ceph mgr code in python is most commonly written by adding mgr
+modules and corresponding classes and then adding methods to those classes that
+are decorated using `@CLICommand` from  `mgr_module.py`.  These methods (that
+will be called endpoints subsequently) then implement the logic that is
+executed when the mgr receives a command from a client.  These endpoints are
+currently responsible for forming a response tuple of (int, str, str) where the
+int represents a return value (error code) and the first string the "body" of
+the response. The mgr supports a generic `format` parameter (`--format` on the
+ceph cli) that each endpoint must then explicitly handle. At the time of this
+writing, many endpoints do not handle alternate formats and are each
+implementing formatting/serialization of values in various different ways.
+
+The `object_format` module aims to make the process of writing endpoint
+functions easier, more consistent, and (hopefully) better documented.  At the
+highest level, the module provides a new decorator `Responder` that must be
+placed below the `CLICommand` decorator (so that it decorates the endpoint
+before `CLICommand`). This decorator helps automatically convert Python objects
+to response tuples expected by the manager, while handling the `format`
+parameter automatically.
+
+In addition to the decorator the module provides a few other types and methods
+that intended to interoperate with the decorator and make small customizations
+and error handling easier.
+
+== Using Responder ==
+
+The simple and intended way to use the decorator is as follows:
+    @CLICommand("command name", perm="r")
+    Responder()
+    def create_something(self, name: str) -> Dict[str, str]:
+        ...  # implementation
+        return {"name": name, "id": new_id}
```

Comment:
what are the limitations on what things we can return while using `Responder`? I see they need to be `Python objects` but I'm not totally clear on what that means

Reply:
I'll try to clarify that in the next version of the doc string.